pandas/core/reshape/merge.py

```diff
@@ -921,19 +921,25 @@ def _get_join_info(self):
                         self.left.index,
                         self.right.index,
                         left_indexer,
-                        right_indexer,
                         how="right",
                     )
                 else:
                     join_index = self.right.index.take(right_indexer)
                     left_indexer = np.array([-1] * len(join_index))
             elif self.left_index:
-                if len(self.right) > 0:
+                if self.how == "asof":
```

Comment:
Could you write a comment here explaining why we're doing `how="left"` here?

Reply:
Done